### PR TITLE
Consolidated 'workspace' and 'package' terminology

### PR DESCRIPTION
**Pull request by mattpocock**
This PR removes references to 'packages' when the writer actually meant 'workspaces'. Refocusing this language makes things a lot clearer:

- Workspace: anything containing a `package.json`
- Package: something you get from/deploy to the NPM registry

### CODE REVIEW

**Review by nathanhammond — APPROVED** [review]
I originally wasn't super-excited about `workspace` everywhere, but I think it makes sense as we head toward polyglot monorepos.

The primary key for me is consistency, which this accomplishes.

File: docs/pages/docs/core-concepts/caching.mdx [hunk 1]
```diff
@@ -190,11 +190,11 @@ Note that this automatic detection and inclusion only works if Turborepo success
   cache. To workaround this, Turborepo uses a `TURBO_CI_VENDOR_ENV_KEY` variable to
   _exclude_ environment variables from Turbo's inference.
 
-  For example, Vercel sets the `NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA`. This value changes on every build,
-  so Vercel _also_ sets `TURBO_CI_VENDOR_ENV_KEY="NEXT_PUBLIC_VERCEL_"` to exclude these variables.
+For example, Vercel sets the `NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA`. This value changes on every build,
+so Vercel _also_ sets `TURBO_CI_VENDOR_ENV_KEY="NEXT_PUBLIC_VERCEL_"` to exclude these variables.
```
Comment (mehulkar) [inline docs/pages/docs/core-concepts/caching.mdx:194]: unindent here seems unintentional? it's a `<Callout>` block

**Review by mehulkar — APPROVED** [review]
👏🏾 👏🏾 👏🏾 

- We should relate this somehow in the work towards "single-workspace" repos. No longer calling subdirectories "packages" goes a long way towards building a mental model of how Turborepo operates on _workspaces_, not packages and applications.
- I like that a search for "workspace" in the search takes you to https://turborepo.org/docs/guides/workspaces.Next step may be to consolidate the idea in that doc. 
- We probably don't care, but do we have any way to measure SEO impact? The only thing I can *really* tell is that https://turborepo.org/docs/core-concepts/filtering changed its page title from "Filtering Packages | Turborepo" to "Filtering Workspaces | Turborepo" (changing frontmatter). I would be curious to see how our inbound search traffic changes over time as we make these content changes.

File: docs/pages/docs/core-concepts/caching.mdx (hunk 1 above)
Comment (mattpocock) [reply docs/pages/docs/core-concepts/caching.mdx:194]: Prettier is telling me this is OK, so I'll trust it.

**Comment by mattpocock** [comment]
@jaredpalmer Done

**Pull request merged**